### ISSUE TITLE
Exception to structure.join_condition_order for range joins

### ISSUE THREAD

**Issue by RoyalTS** [sqlfluff/sqlfluff#6024]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

I generally agree that join conditions ought to have the left-hand table on the left side of the condition, like this example from the docs:

```sql
select
    foo.a,
    foo.b,
    bar.c
from foo
left join bar
    on foo.a = bar.a
    and foo.b = bar.b
```

However, there is one case where breaking with this rule aids readability and understanding: range joins:

```sql
select
    foo.event_at,
    foo.b,
    bar.c
from foo
left join bar
    on bar.start_at <= foo.event_at
    and foo.event_at < bar.end_at
```

Would it be possible to allow for this exception, at least as an option to `structure.join_condition_order`?

### Dialect

All

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by thibonacci** [comment]
Hey @RoyalTS, the nice thing about this rule is that it's very simple: list the table referenced earlier - or later - first.
IMO what you're asking is too specific and sort of would set a precedent that I'm not comfortable with. In addition it's still subjective, I don't think everybody will agree with the order you prefer when it comes to ranges.
I would recommend to use [inline error ignoring](https://docs.sqlfluff.com/en/stable/rules.html#inline-ignoring-errors) whenever you're not satisfied with this rule.

**Comment by peterbud** [comment]
I agree wit @thibonacci . Using the rule exclusion such personal preferences can be handled in a flexible way.
